Safewaan/HonkedIn
# Implemented search function in forums and resources

## Trello ticket link

[As a user, I can search for forums so that I can find relevant topics.](https://trello.com/c/K5XlVZev )
[As a user, I can search for resources so that I can find relevant resources.](https://trello.com/c/gMZmEUcw)



## Implementation description
* Added search functionality to Forums, My Forums, and Resources page
* For the Forum page, users can search by forum title, description, and creator name 
* For the My Forums page, users can search by forum title and description
* For the Resouces page, users can search by resource title and creator name
* Updated styling for search bar, search button, and clear search
* Added Clear Search functionality to the Search.js file



## Steps to test
1. If you have not already installed the new styling components, run `npm i`
2. Go to each of the pages: Forums, My Forums, and Resources
3. Ensure that when the page renders, the entire list of forums/resources is there
4. Test that you can type any string into the text box
5. Clicking the magnifying glass icon should run the search and the list of items that match your search should be rendered
6. Check that the returned list matches your search term 
7. If the search bar is empty, the page should return the entire list of items  in the db 
8. Clicking the "Clear Search" should reset the search bar and render the list of all items in the db



## What should reviewers focus on?
* Check the styling is consistent
* Ensure that the search function works as expected
